services/workflows-service/src/transaction/dtos/transaction-create.dto.ts

```diff
@@ -31,16 +30,14 @@ import { EndUserCreateDto } from '@/end-user/dtos/end-user-create';
 
 export class CounterpartyInfo {
   @ApiProperty({ required: true }) @IsString() correlationId!: string;
-  @ApiProperty({ required: false }) @IsString() @IsOptional() id?: string;
-  @ApiProperty({ required: true }) @IsString() @IsOptional() type?: CounterpartyType;
   @ApiProperty({ required: false }) @IsString() @IsOptional() sortCode?: string;
   @ApiProperty({ required: false }) @IsString() @IsOptional() bankCountry?: string;
 
-  @ApiProperty({ required: false })
-  @IsString()
-  @Type(() => EndUserCreateDto)
   @IsOptional()
-  entityData?: (BusinessCreateDto | EndUserCreateDto) & { id: string };
+  businessData?: Omit<BusinessCreateDto, 'correlationId'>;
+
+  @IsOptional()
+  endUserData?: Omit<EndUserCreateDto, 'correlationId'>;
```

Comment:
Both are optional?
Dont we want to declare that one of them have to present?

Reply:
As I understood, both can be optional, and I wrote a test for it. I'll verify it with Alon.